src/libraries/System.Runtime.Serialization.Schema/ref/System.Runtime.Serialization.Schema.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the https://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System.Runtime.Serialization
+{
+    public partial interface ISerializationCodeDomSurrogateProvider
+    {
+        System.CodeDom.CodeTypeDeclaration ProcessImportedType(System.CodeDom.CodeTypeDeclaration typeDeclaration, System.CodeDom.CodeCompileUnit compileUnit);
+    }
+    public partial class ImportOptions
+    {
+        public System.CodeDom.Compiler.CodeDomProvider? CodeProvider { get { throw null; } set { throw null; } }
+        public bool EnableDataBinding { get { throw null; } set { throw null; } }
+        public ISerializationSurrogateProvider? DataContractSurrogate { get { throw null; } set { throw null; } }
+        public bool GenerateInternal { get { throw null; } set { throw null; } }
+        public bool GenerateSerializable { get { throw null; } set { throw null; } }
+        public bool ImportXmlType { get { throw null; } set { throw null; } }
+        public System.Collections.Generic.IDictionary<string, string> Namespaces { get { throw null; } }
+        public System.Collections.Generic.ICollection<Type> ReferencedCollectionTypes { get { throw null; } }
+        public System.Collections.Generic.ICollection<Type> ReferencedTypes { get { throw null; } }
+    }
+    public partial class XsdDataContractImporter
+    {
+        public XsdDataContractImporter() { throw null; }
+        public XsdDataContractImporter(System.CodeDom.CodeCompileUnit codeCompileUnit) { throw null; }
+
+        public System.CodeDom.CodeCompileUnit CodeCompileUnit { get { throw null; } }
+        public ImportOptions? Options { get { throw null; } set { throw null; } }
+
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public bool CanImport(System.Xml.Schema.XmlSchemaSet schemas) { throw null; }
```

Comment:
Do we need this on this assembly? As long as S.RT.Serialization isn't trimmed, isn't this code safe to trim based on static analysis?

Reply:
Sys.RT.Serialization stuff can be trimmed. This attribute means that this code "references" other code which might get trimmed, so the trimmer can produce a warning if your code uses this method. So I do believe these attributes are necessary in general. (It might be worth looking through them to see if we can reduce their propagation through everything, but that's a different question.)